src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -91,12 +95,61 @@ public override async Task<IReadOnlyList<PackageSpec>> GetPackageSpecsAsync(Depe
         /// </summary>
         public override async Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
-            ProjectPackages packages = await GetInstalledAndTransitivePackagesAsync(includeTransitiveOrigins: false, token);
-            return packages.InstalledPackages;
+            token.ThrowIfCancellationRequested();
+
+            (PackageSpec packageSpec, string assetsFilePath) = await GetCurrentPackageSpecAndAssetsFilePathSafeAsync(token);
+
+            if (packageSpec == null) // null means project is not nominated
+            {
+                IsInstalledComputationNeeded = true;
+                return Array.Empty<PackageReference>();
+            }
+
+            IReadOnlyList<LockFileTarget> targetsList = null;
+            T installedPackages;
+            if (IsInstalledComputationNeeded)
+            {
+                installedPackages = new T();
+                targetsList = (await GetTargetsListAsync(assetsFilePath, token))?.ToList();
```

Comment:
Just curious, do we need `.ToList();` here?

Reply:
Good catch. Another API requires IReadOnlyList, I've thining on chaning that API to avoid this call.

Reply:
Removed all `.ToList()` calls in `GetTargetsListAsync()`